src/store/TimeperiodStore.js

```diff
@@ -38,6 +38,8 @@ export default class TimeperiodStore {
         ], this.updateCountdown.bind(this));
 
         this.context.stx.addEventListener('newChart', this.updateDisplay);
+
+        reaction(() => this.mainStore.state.granularity, granularity => this.onGranularityChange(granularity));
```

Comment:
@shayan-binary The reaction will cause an exception if `onGranularityChange` is not assigned for `Timeperiod` component. it would be better to check `onGranularityChange` to be a function before invoking it or assign a default value to it at line 22 (e.g `onGranularityChange = () => null`